### PR TITLE
Task 5

### PR DESCRIPTION
**Pull request by SamoylenkVi**

### CODE REVIEW

File: src/presenter/movie-wrapper.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
+import EmptyFilmMessageView from '../view/message-empty-film-list';
```
Comment (Pavel-Liteiniy) [inline src/presenter/movie-wrapper.js]: Change file name to 'movie-list'

File: src/presenter/movie-wrapper.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import ShowMoreButtonView from '../view/show-more-button';
+import MovieWrapperView from '../view/movie-wrapper';
+import MovieCardPresenter from './movie';
+
+import {
+  renderElement, RenderPosition, remove,
```
Comment (Pavel-Liteiniy) [inline src/presenter/movie-wrapper.js]: Is `RenderPosition` constant? Maybe will be better to move it to /constants?

File: src/presenter/movie-wrapper.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    this._renderFilmsList(filmCards);
+  }
+
+  sortFilms(films) {
+    this._clearFilmsList();
+    this._renderFilmsList(films);
+  }
+
+  _handleFilmUpdate(updateFilm) {
```
Comment (Pavel-Liteiniy) [inline src/presenter/movie-wrapper.js]: If `updateFilm` is not a function then should rename it

```suggestion
  _handleFilmUpdate(updatedFilm) {
```

File: src/presenter/movie-wrapper.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    this._filmsPresenter[updateFilm.id].init(updateFilm);
+  }
+
+  _handleModeChange() {
+    Object
+      .values(this._filmsPresenter)
+      .forEach((presenter) => presenter.resetView());
+  }
+
+  _renderFilmsList(films) {
```
Comment (Pavel-Liteiniy) [inline src/presenter/movie-wrapper.js]:
```suggestion
  _renderFilmList(films) {
```

or

```suggestion
  _renderFilms(films) {
```

File: src/presenter/movie-wrapper.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+      this._movieWrapperList,
+      this._handleFilmUpdate,
+      this._handleModeChange,
+    );
+
+    filmPresenter.init(film);
+    this._filmsPresenter[film.id] = filmPresenter;
+  }
+
+  _renderCardsFilm(from, to) {
```
Comment (Pavel-Liteiniy) [inline src/presenter/movie-wrapper.js]:
```suggestion
  _renderCardFilms(from, to) {
```

File: src/presenter/movie-wrapper.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+    this._showMoreButton.setClickHandler(this._handleLoadMoreButtonClick);
+  }
+
+  _renderFilmsContainer() {
+    renderElement(this._filmContainer, this._movieWrapper, RenderPosition.BEFOREEND);
+    this._movieWrapperList = this._movieWrapper.getFilmsListContainer();
+  }
+
+  _clearFilmsList() {
```
Comment (Pavel-Liteiniy) [inline src/presenter/movie-wrapper.js]:
```suggestion
  _clearFilmList() {
```

File: src/view/film-details.js [hunk 7]
```diff
@@ -190,4 +191,30 @@ export default class FilmCardDetails extends AbstractView {
...
+      isAddedToWatchList: evt.target.hasAttribute('data-watchlist'),
+      isAddedToWatched: evt.target.hasAttribute('data-watched'),
+      isAddedToFavorite: evt.target.hasAttribute('data-favorite'),
+    });
+  }
+
+  setSpecialListHandler(callback) {
+    this._callback.specialList = callback;
+
+    this.getElement().querySelector('.film-details__control-button--watchlist').addEventListener(
```
Comment (Pavel-Liteiniy) [inline src/view/film-details.js]: Can we add here only one EventListener?

File: src/view/movie-card.js [hunk 8]
```diff
@@ -48,24 +48,48 @@ export default class FilmCard extends AbstractView {
...
+      isAddedToWatchList: evt.target.hasAttribute('data-watchlist'),
+      isAddedToWatched: evt.target.hasAttribute('data-watched'),
+      isAddedToFavorite: evt.target.hasAttribute('data-favorite'),
+    });
+  }
+
+  setSpecialListHandler(callback) {
+    this._callback.specialList = callback;
+
+    this.getElement().querySelector('.film-card__controls-item--add-to-watchlist').addEventListener(
```
Comment (Pavel-Liteiniy) [inline src/view/movie-card.js]: Can we add here only one EventListener?

File: src/view/sort.js [hunk 9]
```diff
@@ -2,13 +2,47 @@ import AbstractView from './abstract';
...
+
+    const allButtons = this.getElement().querySelectorAll('.sort__button');
+    allButtons.forEach((button) => button.classList.remove('sort__button--active'));
+    evt.target.classList.add('sort__button--active');
+  }
+
+  setSortMovieHandler(callback) {
+    this._callback.sortMovie = callback;
+
+    this.getElement().querySelector('[data-sort-default]').addEventListener(
```
Comment (Pavel-Liteiniy) [inline src/view/sort.js]: Can we add here only one EventListener?

Comment (SamoylenkVi) [reply src/view/sort.js]: fixed

File: src/view/movie-card.js (hunk 8 above)
Comment (SamoylenkVi) [reply src/view/movie-card.js]: fixed

File: src/view/film-details.js (hunk 7 above)
Comment (SamoylenkVi) [reply src/view/film-details.js]: fixed

File: src/presenter/movie-wrapper.js (hunk 6 above)
Comment (SamoylenkVi) [reply src/presenter/movie-wrapper.js]: fixed

File: src/presenter/movie-wrapper.js (hunk 5 above)
Comment (SamoylenkVi) [reply src/presenter/movie-wrapper.js]: fixed

File: src/presenter/movie-wrapper.js (hunk 4 above)
Comment (SamoylenkVi) [reply src/presenter/movie-wrapper.js]: fixed

File: src/presenter/movie-wrapper.js (hunk 3 above)
Comment (SamoylenkVi) [reply src/presenter/movie-wrapper.js]: fixed

File: src/presenter/movie-wrapper.js (hunk 1 above)
Comment (SamoylenkVi) [reply src/presenter/movie-wrapper.js]: fixed

File: src/presenter/movie-wrapper.js (hunk 2 above)
Comment (SamoylenkVi) [reply src/presenter/movie-wrapper.js]: fixed

**Comment by Pavel-Liteiniy** [comment]
Approved

**Pull request merged**